cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.kt

```diff
@@ -240,68 +242,69 @@ class CXXLanguageFrontend(config: TranslationConfiguration, scopeManager: ScopeM
                         return null
                     }
 
-                // Get start column by stepping backwards from begin of node to first occurrence of
-                // '\n'
-                var startColumn = 1
-                for (i in node.fileLocation.nodeOffset - 1 downTo 2) {
-                    if (i >= translationUnitRawSignature.length) {
-                        // Fail gracefully, so that we can at least find out why this fails
-                        LOGGER.warn(
-                            "Requested index {} exceeds length of translation unit code ({})",
-                            i,
-                            translationUnitRawSignature.length
-                        )
-                        return null
-                    }
-                    if (translationUnitRawSignature[i] == '\n') {
-                        break
+                // our start line, indexed by 0
+                val startLine = node.fileLocation.startingLineNumber - 1
+
+                // our end line, indexed by 0
+                val endLine = node.fileLocation.endingLineNumber - 1
+
+                // our start column, index by 0
+                val startColumn =
+                    if (startLine == 0) {
+                        // if we are in the first line, the start column is just the node offset
+                        node.fileLocation.nodeOffset
+                    } else {
+                        // otherwise, we need to calculate the difference to the previous line break
+                        node.fileLocation.nodeOffset - lineBreaks[startLine - 1] - 1
                     }
-                    startColumn++
-                }
+
+                // our end column, index by 0
                 val endColumn =
-                    getEndColumnIndex(
-                        translationUnitRawSignature,
-                        node.fileLocation.nodeOffset + node.length
-                    )
-                val region =
-                    Region(
-                        fLocation.startingLineNumber,
-                        startColumn,
-                        fLocation.endingLineNumber,
-                        endColumn
-                    )
+                    if (endLine == 0) {
+                        // if we are in the first line, the start column is just the node offset
+                        node.fileLocation.nodeOffset + node.fileLocation.nodeLength
+                    } else {
+                        // otherwise, we need to calculate the difference to the previous line break
+                        (node.fileLocation.nodeOffset + node.fileLocation.nodeLength) -
+                            lineBreaks[endLine - 1] -
+                            1
+                    }
+
+                // for a SARIF compliant format, we need to add +1, since its index begins at 1 and
+                // not 0
+                val region = Region(startLine + 1, startColumn + 1, endLine + 1, endColumn + 1)
```

Comment:
I know. I kept this it deliberately this way because calculating the line numbers is confusing stuff, especially if CDT internally sometimes has 0-indexed and sometimes has 1-indexed values. My guess is that the compiler will optimise this any way.